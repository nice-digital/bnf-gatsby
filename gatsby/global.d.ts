// Let TypeScript know we're importing SCSS (and SCSS modules)
// Avoids "Can't import CSS/SCSS modules. TypeScript says “Cannot Find Module”"
// See https://stackoverflow.com/a/41946697/486434
declare module "*.module.scss" {
	const content: { [className: string]: string };
	export default content;
}

// So that we can import JSON files in TypeScript.
declare module "*.json" {
	const value: unknown;

	export default value;
}
