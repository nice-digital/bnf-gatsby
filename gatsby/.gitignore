node_modules/
.cache/
public

# dotenv environment variables file
.env
.env.*
