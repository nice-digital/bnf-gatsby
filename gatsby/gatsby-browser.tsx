import {
	ShouldUpdateScrollArgs,
	type RouteUpdateArgs,
	type WrapPageElementBrowserArgs,
} from "gatsby";
import { type ReactElement } from "react";

import { Layout } from "@/components/Layout/Layout";

// Gatsby hook for when the route has changed on the client side
// See https://www.gatsbyjs.org/docs/browser-apis/#onRouteUpdate
export const onRouteUpdate = ({
	prevLocation,
	location,
}: RouteUpdateArgs): void => {
	if (prevLocation) {
		// Cast is needed because of https://github.com/gatsbyjs/gatsby/issues/29124
		const prevPath = (prevLocation as Location).pathname,
			path = location.pathname;

		// Push our own event to the dataLayer on page change rather than using the
		// 'gatsby-route-change' event built into gatsby-plugin-google-tagmanager.
		// Because gatsby-route-change is pushed on initial page load as well as route change,
		// AND because it doesn't use requestAnimationFrame to delay until the page title has been updated

		const sendPageView = () => {
			window.dataLayer.push({
				location: location.href,
				referrer: prevLocation.href,
			});
			// Don't consider hash changes to be a page view - pageviews only happy when the path changes
			if (prevPath != path) window.dataLayer.push({ event: "pageview" });
		};

		// Delay before push to the data layer, to make sure the page title has been updated
		// See https://github.com/gatsbyjs/gatsby/pull/10917/files#diff-bf0d94c8bf47d5c1687e342c2dba1e00R12-R13
		if ("requestAnimationFrame" in window) {
			window.requestAnimationFrame(() => {
				window.requestAnimationFrame(sendPageView);
			});
		} else {
			// simulate 2 rAF calls
			setTimeout(sendPageView, 32);
		}
	} else {
		window.dataLayer.push({
			location: location.href,
			referrer: document.referrer,
		});
	}
	// Default return statement when prevLocation is falsy
	return;
};

export const wrapPageElement = ({
	element,
	props,
}: WrapPageElementBrowserArgs): ReactElement => (
	<Layout {...props}>{element}</Layout>
);

/**
 * Gatsby hook for overriding scroll position
 * See https://www.gatsbyjs.org/docs/browser-apis/#shouldUpdateScroll
 */
export const shouldUpdateScroll = ({
	routerProps: { location },
	prevRouterProps,
	getSavedScrollPosition,
}: ShouldUpdateScrollArgs): boolean | string | [number, number] => {
	if (
		// If there's no previous route props we're coming from an external site, which means
		// we want to scroll to the hash (if there is one), and _not_ a saved scroll position
		!prevRouterProps ||
		// Or we're linking to as hash within the same page
		(prevRouterProps.location.pathname === location.pathname && location.hash)
	) {
		// Provide our own scroll to hash to avoid Gatsby using a stored scroll position
		requestAnimationFrame(() => {
			requestAnimationFrame(() => {
				const targetElement = location.hash
					? document.querySelector(location.hash)
					: null;

				if (targetElement) {
					targetElement.setAttribute("tabIndex", "-1");
					(targetElement as HTMLElement).focus();
					targetElement.scrollIntoView();
				}
			});
		});
		return false;
	}

	const savedScrollY = (getSavedScrollPosition(location)?.[1] || 0) as number;
	if (savedScrollY > 0) {
		window.scrollTo(0, savedScrollY);
		return false;
	}

	if (location.hash) {
		requestAnimationFrame(() => {
			requestAnimationFrame(() => {
				const targetElement = document.querySelector(location.hash);

				// Default to Gatsby's default behaviour if the element doesn't exist
				if (!targetElement) return true;

				// Provide our own scroll to hash to avoid Gatsby using a stored scroll position
				targetElement.setAttribute("tabIndex", "-1");
				(targetElement as HTMLElement).focus();
				targetElement.scrollIntoView();
				return false;
			});
		});
		return false;
	}

	requestAnimationFrame(() => {
		requestAnimationFrame(() => {
			const contentStartElement = document.getElementById("content-start");

			if (!contentStartElement) return true;

			contentStartElement.setAttribute("tabIndex", "-1");

			//scrolling to 0,0 to preserve existing BNF navigation behaviour
			window.scrollTo(0, 0);
			return false;
		});
	});
	// Default to scrolling to the content start element as the standard navigation behaviour

	return true;
};
