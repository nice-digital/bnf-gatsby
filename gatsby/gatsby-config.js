require("dotenv").config({
	path: `.env.${process.env.NODE_ENV}`,
});

const isBNF = process.env.GATSBY_SITE === "bnf";

if (!process.env.GATSBY_SEARCH_URL)
	throw new Error(
		"Env var GATSBY_SEARCH_URL isn't set. Did you forget to add it?"
	);

module.exports = {
	jsxRuntime: "automatic",
	siteMetadata: {
		isBNF,
		siteUrl: `https://${isBNF ? "bnf" : "bnfc"}.nice.org.uk`,
		siteTitleShort: isBNF ? "BNF" : "BNFC",
		siteTitleLong:
			"British National Formulary" + (isBNF ? "" : " for Children"),
	},
	plugins: [
		{
			resolve: `gatsby-source-bnf`,
			options: {
				feedURL: process.env.FEED_URL,
				userKey: process.env.FEED_USER_KEY,
				site: isBNF ? "bnf" : "bnfc",
			},
		},
		// Avoid errors like "ModuleNotFoundError: Module not found: Error: Can't resolve '@/components/Layout/Layout'" when using custom paths in tsconfig.json
		`gatsby-plugin-tsconfig-paths`,
		{
			resolve: `gatsby-plugin-sass`,
			options: {
				// Make CSS modules play nicely with TypeScript, see also global.d.ts for *.scss TS def
				cssLoaderOptions: {
					esModule: false,
					modules: {
						namedExport: false,
					},
				},
			},
		},
		"gatsby-plugin-catch-links",
		"gatsby-plugin-react-helmet",
		"gatsby-plugin-remove-generator",
		{
			resolve: `gatsby-plugin-sitemap`,
			options: {
				excludes: [`/search/`],
			},
		},
		{
			resolve: "gatsby-plugin-google-tagmanager",
			options: {
				id: "GTM-5H5L9GK",
				includeInDevelopment: true,
			},
		},
		{
			resolve: "gatsby-plugin-manifest",
			options: {
				icon: "src/images/icon.png",
			},
		},
	],
};
