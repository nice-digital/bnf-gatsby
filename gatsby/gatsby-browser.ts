import { RouteUpdateArgs } from "gatsby";

// Gatsby hook for when the route has changed on the client side
// See https://www.gatsbyjs.org/docs/browser-apis/#onRouteUpdate
export const onRouteUpdate = ({
	prevLocation,
	location,
}: RouteUpdateArgs): void => {
	if (prevLocation) {
		// Cast is needed because of https://github.com/gatsbyjs/gatsby/issues/29124
		const prevPath = (prevLocation as Location).pathname,
			path = location.pathname;

		// Push our own event to the dataLayer on page change rather than using the
		// 'gatsby-route-change' event built into gatsby-plugin-google-tagmanager.
		// Because gatsby-route-change is pushed on initial page load as well as route change,
		// AND because it doesn't use requestAnimationFrame to delay until the page title has been updated

		const sendPageView = () => {
			window.dataLayer.push({
				location: location.href,
				referrer: prevLocation.href,
			});
			// Don't consider hash changes to be a page view - pageviews only happy when the path changes
			if (prevPath != path) window.dataLayer.push({ event: "pageview" });
		};

		// Delay before push to the data layer, to make sure the page title has been updated
		// See https://github.com/gatsbyjs/gatsby/pull/10917/files#diff-bf0d94c8bf47d5c1687e342c2dba1e00R12-R13
		if ("requestAnimationFrame" in window) {
			window.requestAnimationFrame(() => {
				window.requestAnimationFrame(sendPageView);
			});
		} else {
			// simulate 2 rAF calls
			setTimeout(sendPageView, 32);
		}
	} else {
		window.dataLayer.push({
			location: location.href,
			referrer: document.referrer,
		});
	}
};
