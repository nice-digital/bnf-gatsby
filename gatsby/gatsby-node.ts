import { CreateSchemaCustomizationArgs, CreateWebpackConfigArgs } from "gatsby";

import { initialFieldExtension } from "./src/field-extensions/initial";

/**
 * Gatsby hook for customizing the schema.
 * See https://www.gatsbyjs.org/docs/schema-customization/
 */
export const createSchemaCustomization = ({
	actions: { createFieldExtension, createTypes },
}: CreateSchemaCustomizationArgs): void => {
	createFieldExtension(initialFieldExtension);

	// Add initials for types used in A-Z lists, as it's easier for grouping
	// We don't add the initial field in the source plugin because grouping is a presentation concern
	const typeDefs = `
	  type BnfDrug implements Node {
		"The lowercase first letter of the title, used for grouping"
		initial: String! @initial(field: "title")
	  }
	  type BnfTreatmentSummary implements Node {
		"The lowercase first letter of the title, used for grouping"
		initial: String! @initial(field: "title")
	  }
	  type BnfInteractant implements Node {
		"The lowercase first letter of the title, used for grouping"
		initial: String! @initial(field: "title")
	  }
	`;
	createTypes(typeDefs);
};

/**
 * Gatsby hook for overriding webpack config
 * See https://www.gatsbyjs.com/docs/how-to/custom-configuration/add-custom-webpack-config/
 */
export const onCreateWebpackConfig = ({
	stage,
	actions,
	getConfig,
	plugins,
}: CreateWebpackConfigArgs): void => {
	// Disable order warnings from mini CSS plugin
	// See https://github.com/gatsbyjs/gatsby/discussions/30169#discussioncomment-621285
	if (stage === "build-javascript" || stage === "develop") {
		const config = getConfig();

		const miniCss = config.plugins.find(
			(plugin: () => void) => plugin.constructor.name === "MiniCssExtractPlugin"
		);

		if (miniCss) {
			miniCss.options.ignoreOrder = true;
		}

		actions.replaceWebpackConfig(config);

		actions.setWebpackConfig({
			plugins: [plugins.provide({ process: "process/browser" })],
		});
	}
};
