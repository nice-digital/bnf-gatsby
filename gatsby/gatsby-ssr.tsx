import { PreRenderHTMLArgs } from "gatsby";
import React from "react";

export const onPreRenderHTML = ({
	getHeadComponents,
	replaceHeadComponents,
}: PreRenderHTMLArgs): void => {
	const components = getHeadComponents();

	components.push(
		<script
			src={process.env.GATSBY_COOKIE_BANNER_URL}
			type="text/javascript"
			key="cookie-banner"
			async
		></script>
	);

	replaceHeadComponents(components);
};
