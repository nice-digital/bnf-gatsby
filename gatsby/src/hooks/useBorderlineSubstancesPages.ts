import { useStaticQuery, graphql } from "gatsby";

import { SID, PHPID } from "@nice-digital/gatsby-source-bnf";

import { type MenuPageLink, type SlugAndTitle } from "@/utils";

export interface BorderlineSubstancesQueryResult {
	allBnfBorderlineSubstancesTaxonomyRoot: {
		pages: {
			slug: string;
			title: string;
		}[];
	};
}

const slugToHref = ({ title, slug }: SlugAndTitle): MenuPageLink => ({
	title,
	href: `/borderline-substances/${slug}/`,
});

/**
 * Gets the lists of pages for the borderline substances section
 *
 * @returns The consolidated list of borderline substances pages
 */
export const useBorderlineSubstancesPages = (): MenuPageLink[] => {
	const { allBnfBorderlineSubstancesTaxonomyRoot } =
		useStaticQuery<BorderlineSubstancesQueryResult>(
			graphql`
				{
					allBnfBorderlineSubstancesTaxonomyRoot {
						pages: nodes {
							slug
							title
						}
					}
				}
			`
		);

	return allBnfBorderlineSubstancesTaxonomyRoot.pages.map(slugToHref);
};
