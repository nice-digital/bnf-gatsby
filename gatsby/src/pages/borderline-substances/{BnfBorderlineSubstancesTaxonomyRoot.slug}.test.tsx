import { useLocation } from "@reach/router";
import { render, waitFor, screen, within } from "@testing-library/react";
import { useStaticQuery } from "gatsby";

import { mockBorderlineSubstancesPagesQueryData } from "@/hooks/useBorderlineSubstancesPages.test";

import BorderlineSubstancesRootPage, {
	type BorderlineSubstancesRootPageProps,
	query,
} from "./{BnfBorderlineSubstancesTaxonomyRoot.taxonomy__slug}";

const minimumProps: BorderlineSubstancesRootPageProps = {
	data: {
		bnfBorderlineSubstancesTaxonomyRoot: {
			taxonomy: {
				title: "Specialised formulas",
				slug: "specialised-formulas",
				childTaxonomies: [
					{
						title: "Specialised formulas for specific clinical conditions",
						slug: "specialised-formulas-for-specific-clinical-conditions",
						childTaxonomies: [],
					},
				],
			},
		},
	},
};

describe("BorderlineSubstancesRootPage", () => {
	beforeEach(() => {
		// useStaticQuery is used within the left hand menu (BorderlineSubstancesMenu)
		(useStaticQuery as jest.Mock).mockReturnValue(
			mockBorderlineSubstancesPagesQueryData
		);
	});

	it("should match snapshot for graphql query", () => {
		expect(query).toMatchSnapshot();
	});

	describe("SEO", () => {
		beforeEach(() => {
			document.title = "";
		});

		it("should set page title from borderline substance taxonomy title", async () => {
			render(<BorderlineSubstancesRootPage {...minimumProps} />);

			await waitFor(() => {
				expect(document.title).toStartWith(
					"Specialised formulas | Borderline substances | "
				);
			});
		});
	});

	describe("Page header", () => {
		it("should add content start skip link target id to page header", () => {
			render(<BorderlineSubstancesRootPage {...minimumProps} />);
			const pageHeader = screen.getByTestId("page-header");
			expect(pageHeader).toHaveAttribute("id", "content-start");
		});

		it("should render heading 1 with current page title", () => {
			render(<BorderlineSubstancesRootPage {...minimumProps} />);
			const heading1 = screen.getByRole("heading", {
				level: 1,
			});
			expect(heading1).toHaveTextContent("Specialised formulas");
		});
	});

	describe("Breadcrumbs", () => {
		it.each([
			["NICE", "https://www.nice.org.uk/"],
			["BNF", "/"],
			["Borderline substances", "/borderline-substances/"],
		])(
			"should render default '(%s)' breadcrumb",
			(breadcrumbText, expectedHref) => {
				render(<BorderlineSubstancesRootPage {...minimumProps} />);

				const breadcrumbNav = screen.getByRole("navigation", {
					name: "Breadcrumbs",
				});
				const breadcrumb = within(breadcrumbNav).getByRole("link", {
					name: breadcrumbText,
				});

				expect(breadcrumb).toHaveAttribute("href", expectedHref);
			}
		);

		it("should render current page breadcrumb without link", () => {
			render(<BorderlineSubstancesRootPage {...minimumProps} />);

			const breadcrumbNav = screen.getByRole("navigation", {
				name: "Breadcrumbs",
			});
			const currentPageCrumb = within(breadcrumbNav).getByText(
				"Specialised formulas"
			);
			expect(currentPageCrumb).toBeInTheDocument();
			expect(currentPageCrumb.tagName).toBe("SPAN");
		});
	});

	describe("Body", () => {
		describe("Single level", () => {
			beforeEach(() => {
				(useLocation as jest.Mock).mockImplementation(
					() =>
						new URL(
							"https://bnf-gatsby-tests.nice.org.uk/borderline-substances/specialised-formulas/"
						)
				);
			});

			it("should not render any heading 2s", () => {
				render(<BorderlineSubstancesRootPage {...minimumProps} />);
				expect(
					screen.queryAllByRole("heading", {
						level: 2,
					})
				).toHaveLength(0);
			});

			it("should render labelled list of sub pages", () => {
				render(<BorderlineSubstancesRootPage {...minimumProps} />);
				expect(
					screen.getByRole("list", {
						name: "Sections within Specialised formulas",
					})
				).toBeInTheDocument();
			});

			it("should render links to sub taxonomy pages", () => {
				render(<BorderlineSubstancesRootPage {...minimumProps} />);
				expect(
					screen.getByRole("link", {
						name: "Specialised formulas for specific clinical conditions",
					})
				).toHaveAttribute(
					"href",
					"/borderline-substances/specialised-formulas/specialised-formulas-for-specific-clinical-conditions/"
				);
			});

			it("should match snapshot", () => {
				const { container } = render(
					<BorderlineSubstancesRootPage {...minimumProps} />
				);
				expect(container).toMatchSnapshot();
			});
		});

		describe("Two levels", () => {
			const props: BorderlineSubstancesRootPageProps = {
				data: {
					bnfBorderlineSubstancesTaxonomyRoot: {
						taxonomy: {
							title: "Foods for special diets",
							slug: "foods-for-special-diets",
							childTaxonomies: [
								{
									title: "Gluten-free foods",
									slug: "gluten-free-foods",
									childTaxonomies: [
										{
											title: "Bread",
											slug: "bread",
										},
										{
											title: "Cereals",
											slug: "cereals",
										},
									],
								},
								{
									title: "Low-protein foods",
									slug: "low-protein-foods",
									childTaxonomies: [
										{
											title: "Bread",
											slug: "bread-2",
											childTaxonomies: [],
										},
										{
											title: "Cake, biscuits, and snacks",
											slug: "cake-biscuits-and-snacks",
											childTaxonomies: [],
										},
									],
								},
							],
						},
					},
				},
			};

			beforeEach(() => {
				(useLocation as jest.Mock).mockImplementation(
					() =>
						new URL(
							"https://bnf-gatsby-tests.nice.org.uk/borderline-substances/foods-for-special-diets/"
						)
				);
			});

			it("should render section with accessible name for each sub taxonomy", () => {
				render(<BorderlineSubstancesRootPage {...props} />);
				expect(
					screen.getByRole("region", {
						name: "Gluten-free foods",
					})
				).toBeInTheDocument();
				expect(
					screen.getByRole("region", {
						name: "Low-protein foods",
					})
				).toBeInTheDocument();
			});

			it("should render heading 2 for each sub taxonomy", () => {
				render(<BorderlineSubstancesRootPage {...props} />);
				expect(
					screen.getByRole("heading", {
						level: 2,
						name: "Gluten-free foods",
					})
				).toBeInTheDocument();
				expect(
					screen.getByRole("heading", {
						level: 2,
						name: "Low-protein foods",
					})
				).toBeInTheDocument();
			});

			it("should render list for each sub taxonomy", () => {
				render(<BorderlineSubstancesRootPage {...props} />);
				expect(
					screen.getByRole("list", {
						name: "Sections within Gluten-free foods",
					})
				).toBeInTheDocument();
				expect(
					screen.getByRole("list", {
						name: "Sections within Low-protein foods",
					})
				).toBeInTheDocument();
			});

			it("should render links to sub taxonomy pages", () => {
				render(<BorderlineSubstancesRootPage {...props} />);
				expect(
					screen.getByRole("link", {
						name: "Cereals",
					})
				).toHaveAttribute(
					"href",
					"/borderline-substances/foods-for-special-diets/cereals/"
				);
			});

			it("should match snapshot", () => {
				const { container } = render(
					<BorderlineSubstancesRootPage {...props} />
				);
				expect(container).toMatchSnapshot();
			});
		});
	});
});
