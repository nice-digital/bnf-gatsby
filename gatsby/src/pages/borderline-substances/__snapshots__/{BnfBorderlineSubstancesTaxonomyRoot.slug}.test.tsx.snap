// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BorderlineSubstancesRootPage Body Single level should match snapshot 1`] = `
<div>
  <nav
    aria-label="Breadcrumbs"
    data-component="breadcrumbs"
    role="navigation"
  >
    <ol
      class="breadcrumbs"
    >
      <li
        class="breadcrumbs__crumb"
      >
        <a
          href="https://www.nice.org.uk/"
        >
          NICE
        </a>
      </li>
      <li
        class="breadcrumbs__crumb"
      >
        <a
          href="/"
        >
          BNF
        </a>
      </li>
      <li
        class="breadcrumbs__crumb"
      >
        <a
          href="/borderline-substances/"
        >
          Borderline substances
        </a>
      </li>
      <li
        class="breadcrumbs__crumb"
      >
        <span>
          Specialised formulas
        </span>
      </li>
    </ol>
  </nav>
  <script
    type="application/ld+json"
  >
    {
  "@context": "http://schema.org",
  "@type": "BreadcrumbList",
  "itemListElement": [
    {
      "@type": "ListItem",
      "position": 1,
      "item": {
        "@id": "https://www.nice.org.uk/",
        "name": "NICE"
      }
    },
    {
      "@type": "ListItem",
      "position": 2,
      "item": {
        "@id": "/",
        "name": "BNF"
      }
    },
    {
      "@type": "ListItem",
      "position": 3,
      "item": {
        "@id": "/borderline-substances/",
        "name": "Borderline substances"
      }
    },
    {
      "@type": "ListItem",
      "position": 4,
      "item": {
        "name": "Specialised formulas"
      }
    }
  ]
}
  </script>
  <div
    class="page-header "
    data-component="page-header"
    id="content-start"
  >
    <h1
      class="page-header__heading "
    >
      Specialised formulas
    </h1>
  </div>
  <div
    class="grid grid--loose"
    data-testid="body"
  >
    <div
      class="hide-print"
      data-component="grid-item"
      data-g="12 md:4 lg:3"
    >
       
      <button
        aria-expanded="false"
        aria-label="Expand menu for borderline substances pages"
        class="toggleButton"
        type="button"
      >
        <span>
          Specialised formulas
        </span>
        <svg
          aria-hidden="true"
          class="icon"
          fill="none"
          focusable="false"
          height="1em"
          viewBox="0 0 512 512"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M478.152 174c0 10.091-3.525 18.661-10.576 25.712l-186 186c-7.243 7.243-15.909 10.864-26 10.864-10.283 0-18.853-3.621-25.712-10.864l-186-186C36.621 192.853 33 184.283 33 174c0-10.091 3.621-18.757 10.864-26L65 126.576C72.424 119.525 81.09 116 91 116c10.091 0 18.661 3.525 25.712 10.576L255.576 265.44 394.44 126.576c7.051-7.051 15.621-10.576 25.712-10.576 9.909 0 18.576 3.525 26 10.576L467.576 148c7.051 7.424 10.576 16.091 10.576 26Z"
            fill="currentColor"
          />
        </svg>
      </button>
      <nav
        aria-label="Borderline substances pages"
        class="stacked-nav"
        data-component="stacked-nav"
        id="collapsible-menu"
      >
        <p
          class="stacked-nav__root"
        >
          <a
            aria-current="false"
            href="/borderline-substances/"
          >
            Borderline substances
          </a>
        </p>
        <ul
          class="stacked-nav__list"
        >
          <li
            class="stacked-nav__list-item"
          >
            <a
              aria-current="false"
              href="/borderline-substances/enteral-feeds-non-disease-specific/"
            >
              <span
                class="stacked-nav__content-wrapper"
              >
                <span>
                  Enteral feeds (non-disease specific)
                </span>
              </span>
            </a>
          </li>
          <li
            class="stacked-nav__list-item"
          >
            <a
              aria-current="true"
              href="/borderline-substances/specialised-formulas/"
            >
              <span
                class="stacked-nav__content-wrapper"
              >
                <span>
                  Specialised formulas
                </span>
              </span>
            </a>
          </li>
          <li
            class="stacked-nav__list-item"
          >
            <a
              aria-current="false"
              href="/borderline-substances/foods-for-special-diets/"
            >
              <span
                class="stacked-nav__content-wrapper"
              >
                <span>
                  Foods for special diets
                </span>
              </span>
            </a>
          </li>
        </ul>
      </nav>
    </div>
    <div
      data-component="grid-item"
      data-g="12 md:8 lg:9"
    >
      <ol
        aria-label="Sections within Specialised formulas"
        class="column-list column-list--boxed column-list--two-columns"
        data-component="column-list--boxed"
      >
        <li>
          <a
            href="/borderline-substances/specialised-formulas/specialised-formulas-for-specific-clinical-conditions/"
          >
            Specialised formulas for specific clinical conditions
          </a>
        </li>
      </ol>
    </div>
  </div>
</div>
`;

exports[`BorderlineSubstancesRootPage Body Two levels should match snapshot 1`] = `
<div>
  <nav
    aria-label="Breadcrumbs"
    data-component="breadcrumbs"
    role="navigation"
  >
    <ol
      class="breadcrumbs"
    >
      <li
        class="breadcrumbs__crumb"
      >
        <a
          href="https://www.nice.org.uk/"
        >
          NICE
        </a>
      </li>
      <li
        class="breadcrumbs__crumb"
      >
        <a
          href="/"
        >
          BNF
        </a>
      </li>
      <li
        class="breadcrumbs__crumb"
      >
        <a
          href="/borderline-substances/"
        >
          Borderline substances
        </a>
      </li>
      <li
        class="breadcrumbs__crumb"
      >
        <span>
          Foods for special diets
        </span>
      </li>
    </ol>
  </nav>
  <script
    type="application/ld+json"
  >
    {
  "@context": "http://schema.org",
  "@type": "BreadcrumbList",
  "itemListElement": [
    {
      "@type": "ListItem",
      "position": 1,
      "item": {
        "@id": "https://www.nice.org.uk/",
        "name": "NICE"
      }
    },
    {
      "@type": "ListItem",
      "position": 2,
      "item": {
        "@id": "/",
        "name": "BNF"
      }
    },
    {
      "@type": "ListItem",
      "position": 3,
      "item": {
        "@id": "/borderline-substances/",
        "name": "Borderline substances"
      }
    },
    {
      "@type": "ListItem",
      "position": 4,
      "item": {
        "name": "Foods for special diets"
      }
    }
  ]
}
  </script>
  <div
    class="page-header "
    data-component="page-header"
    id="content-start"
  >
    <h1
      class="page-header__heading "
    >
      Foods for special diets
    </h1>
  </div>
  <div
    class="grid grid--loose"
    data-testid="body"
  >
    <div
      class="hide-print"
      data-component="grid-item"
      data-g="12 md:4 lg:3"
    >
       
      <button
        aria-expanded="false"
        aria-label="Expand menu for borderline substances pages"
        class="toggleButton"
        type="button"
      >
        <span>
          Foods for special diets
        </span>
        <svg
          aria-hidden="true"
          class="icon"
          fill="none"
          focusable="false"
          height="1em"
          viewBox="0 0 512 512"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M478.152 174c0 10.091-3.525 18.661-10.576 25.712l-186 186c-7.243 7.243-15.909 10.864-26 10.864-10.283 0-18.853-3.621-25.712-10.864l-186-186C36.621 192.853 33 184.283 33 174c0-10.091 3.621-18.757 10.864-26L65 126.576C72.424 119.525 81.09 116 91 116c10.091 0 18.661 3.525 25.712 10.576L255.576 265.44 394.44 126.576c7.051-7.051 15.621-10.576 25.712-10.576 9.909 0 18.576 3.525 26 10.576L467.576 148c7.051 7.424 10.576 16.091 10.576 26Z"
            fill="currentColor"
          />
        </svg>
      </button>
      <nav
        aria-label="Borderline substances pages"
        class="stacked-nav"
        data-component="stacked-nav"
        id="collapsible-menu"
      >
        <p
          class="stacked-nav__root"
        >
          <a
            aria-current="false"
            href="/borderline-substances/"
          >
            Borderline substances
          </a>
        </p>
        <ul
          class="stacked-nav__list"
        >
          <li
            class="stacked-nav__list-item"
          >
            <a
              aria-current="false"
              href="/borderline-substances/enteral-feeds-non-disease-specific/"
            >
              <span
                class="stacked-nav__content-wrapper"
              >
                <span>
                  Enteral feeds (non-disease specific)
                </span>
              </span>
            </a>
          </li>
          <li
            class="stacked-nav__list-item"
          >
            <a
              aria-current="false"
              href="/borderline-substances/specialised-formulas/"
            >
              <span
                class="stacked-nav__content-wrapper"
              >
                <span>
                  Specialised formulas
                </span>
              </span>
            </a>
          </li>
          <li
            class="stacked-nav__list-item"
          >
            <a
              aria-current="true"
              href="/borderline-substances/foods-for-special-diets/"
            >
              <span
                class="stacked-nav__content-wrapper"
              >
                <span>
                  Foods for special diets
                </span>
              </span>
            </a>
          </li>
        </ul>
      </nav>
    </div>
    <div
      data-component="grid-item"
      data-g="12 md:8 lg:9"
    >
      <section
        aria-labelledby="gluten-free-foods"
      >
        <h2
          class="firstHeading"
          id="gluten-free-foods"
        >
          Gluten-free foods
        </h2>
        <ol
          aria-label="Sections within Gluten-free foods"
          class="column-list column-list--boxed column-list--two-columns"
          data-component="column-list--boxed"
        >
          <li>
            <a
              href="/borderline-substances/foods-for-special-diets/bread/"
            >
              Bread
            </a>
          </li>
          <li>
            <a
              href="/borderline-substances/foods-for-special-diets/cereals/"
            >
              Cereals
            </a>
          </li>
        </ol>
      </section>
      <section
        aria-labelledby="low-protein-foods"
      >
        <h2
          id="low-protein-foods"
        >
          Low-protein foods
        </h2>
        <ol
          aria-label="Sections within Low-protein foods"
          class="column-list column-list--boxed column-list--two-columns"
          data-component="column-list--boxed"
        >
          <li>
            <a
              href="/borderline-substances/foods-for-special-diets/bread-2/"
            >
              Bread
            </a>
          </li>
          <li>
            <a
              href="/borderline-substances/foods-for-special-diets/cake-biscuits-and-snacks/"
            >
              Cake, biscuits, and snacks
            </a>
          </li>
        </ol>
      </section>
    </div>
  </div>
</div>
`;

exports[`BorderlineSubstancesRootPage should match snapshot for graphql query 1`] = `
"
	query ($id: String) {
		bnfBorderlineSubstancesTaxonomyRoot(id: { eq: $id }) {
			taxonomy {
				slug
				title
				childTaxonomies {
					title
					slug
					childTaxonomies {
						title
						slug
						childTaxonomies {
							title
							slug
						}
					}
				}
			}
		}
	}
"
`;
